lib/handler/mruby.c

```diff
@@ -289,13 +324,10 @@ static void handle_exception(h2o_mruby_context_t *ctx, h2o_mruby_generator_t *ge
         fprintf(stderr, "mruby raised: %s\n", RSTRING_PTR(mrb_inspect(mrb, mrb_obj_value(mrb->exc))));
     } else {
         assert(generator->req != NULL);
+        assert(generator->req->_generator == NULL);
```

Comment:
Are we sure that this assertion is always valid?

I believe that a jump from https://github.com/h2o/h2o/blob/af4586f9e7980330bf2b5c2bcf7a967f92eb132d/lib/handler/mruby.c#L1067 has the value set to non-NULL.

Reply:
This assertion may be still valid because L1067 shouldn't be called (i.e. if called, it's our programming miss). So, at https://github.com/h2o/h2o/pull/1217/commits/1986bd62065ed603af84bbb524b6b78f75ba95ea,  I removed that code path, as well as making `handle_exception` more robust because it's called from several locations.